src/credentials/credential-wallet.ts

```diff
@@ -266,101 +266,23 @@ export class CredentialWallet implements ICredentialWallet {
     if (!schema.$metadata.uris['jsonLdContext']) {
       throw new Error('jsonLdContext is missing is the schema');
     }
-    request.context = request.context ?? [];
+    // do copy of request to avoid mutation
+    const r = { ...request };
+    r.context = r.context ?? [];
     if (
-      request.displayMethod?.type === DisplayMethodType.Iden3BasicDisplayMethodV1 &&
-      !request.context.includes(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD)
+      r.displayMethod?.type === DisplayMethodType.Iden3BasicDisplayMethodV1 &&
+      !r.context.includes(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD)
     ) {
-      request.context.push(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD);
+      r.context.push(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD);
     }
-    const context = [
-      VerifiableConstants.JSONLD_SCHEMA.W3C_CREDENTIAL_2018,
-      ...request.context,
-      VerifiableConstants.JSONLD_SCHEMA.IDEN3_CREDENTIAL,
-      schema.$metadata.uris['jsonLdContext']
-    ];
+    r.context.push(schema.$metadata.uris['jsonLdContext']);
+    r.expiration = !r.expiration || r.expiration == 0 ? undefined : r.expiration * 1000;
+    r.id = r.id ? r.id : `urn:${uuid.v4()}`;
```

Comment:
```suggestion
    r.id = r.id ?? `urn:${uuid.v4()}`;
```

Reply:
I tested this on a browser console and there is a result:
```js
const foo = ""; foo ?? console.log(1);
returns: '' - empty string
const foo = ""; foo ? console.log(2) : console.log(1);
returns 1;
```
I expect with user pass empty string for the id field, we need generate uuid.

Reply:
In that case replace ?? to ||   for example a = ''; b = '1'; a = a || b; // '1'